ggshield/core/file_utils.py

```diff
@@ -87,22 +88,41 @@ def get_filepaths(
     return targets
 
 
+def is_path_binary(path: str) -> bool:
+    _, ext = os.path.splitext(path)
+    # `[1:]` because `ext` starts with a "." but extensions in `BINARY_EXTENSIONS` do not
+    return ext[1:] in BINARY_EXTENSIONS
+
+
 def generate_files_from_paths(paths: Iterable[str], verbose: bool) -> Iterator[File]:
     """Loop on filepaths and return an iterator on scannable files."""
     for path in paths:
         if os.path.isdir(path) or not os.path.exists(path):
             continue
 
+        if is_path_binary(path):
+            if verbose:
+                click.echo(
+                    f"ignoring binary file: {path}",
```

Comment:
Nit : Shouldn't we keep the wording "binary file extension" to highlight the fact that the decision is based on the extension ?